core/src/main/scala/kafka/server/ForwardingManager.scala

```diff
@@ -77,7 +79,7 @@ class ForwardingManagerImpl(
 
   override def forwardRequest(
     request: RequestChannel.Request,
-    responseCallback: AbstractResponse => Unit
+    responseCallback: Either[AbstractResponse, Errors] => Unit
```

Comment:
Since we are not using the `Error` from the result, maybe `Option[AbstractResponse]` would be a better type.

Reply:
I'm inclined to keep it as it is, since the current return type gives caller a root cause message.

Reply:
Yeah, but that suggests a generality to the API that doesn't exist. There is only one error that is possible to be returned here.